melange.yaml

```diff
@@ -10,25 +10,23 @@ package:
       - bubblewrap
       - ca-certificates-bundle
 
-environment:
-  contents:
-    packages:
-      - busybox
-      - ca-certificates-bundle
-      - go
-
 pipeline:
   - uses: git-checkout
     with:
       expected-commit: 96097ac9fb328932609073fc37fde9c6ea0b2630
       repository: https://github.com/chainguard-dev/melange
       tag: v${{package.version}}
 
-  - runs: |
-      make melange
-      install -m755 -D ./melange "${{targets.contextdir}}"/usr/bin/melange
-
-  - uses: strip
+  - uses: go/build
+    with:
+      output: melange
+      packages: .
+      ldflags: |
+        -buildid= \
+        -X sigs.k8s.io/release-utils/version.gitVersion=$(git describe --dirty --tags --always) \
+        -X sigs.k8s.io/release-utils/version.gitCommit=$(git rev-parse HEAD) \
+        -X sigs.k8s.io/release-utils/version.gitTreeState=$(shell git diff --quiet >/dev/null 2>&1 || echo "dirty" || echo "clean") \
+        -X sigs.k8s.io/release-utils/version.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
```

Comment:
we should add -d SOURCE_EPOCH_DATE for reproducibility here.

Reply:
updated